DXMainClient/Startup.cs

```diff
@@ -294,6 +310,12 @@ private void CheckSystemSpecifications()
         /// </summary>
         private static void GenerateOnlineId()
         {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Connection.SetId(new Random().Next(int.MaxValue - 1).ToString());
+                return;
+            }
```

Comment:
This is bad, basically Linux players would be unbannable. Need to deprecate Windows registry keys and just use AppData and Linux alternative IMO. cc @Rampastring @Starkku

Reply:
I suggested the lightweight [Microsoft.EntityFrameworkCore.Sqlite](https://docs.microsoft.com/en-us/dotnet/standard/data/sqlite/?tabs=netcore-cli) as registry replacement to have only 1 implementation/location for all platforms. But the data won't be shared between clients then.

Reply:
Is there a need to drag an SQLite DB just for that? We just need some way of storing stuff that is not registry, that's all. File storage in AppData and the equivalent in Linux would work fine IMO.

Reply:
Let's take a step back and look at what Windows specific stuff we're using and for what purpose.

# DisplayOptionsPanel

## DTA/TI/TS Compatibility Fix
- Registry (Microsoft.Win32)
- Windows application sdbinst.exe

## FinalSun Compatibility Fix
- Registry (Microsoft.Win32)
- Windows application sdbinst.exe

# Startup

## Log Hardware info
- WMI (System.Management)

## Generate OnlineId
- Registry (Microsoft.Win32)
- SecurityIdentifier (System.Security.Principal.Windows)
- WMI (System.Management)

All of the above listed items are Windows specific. My conclusion is that none of the above functionality will work on Linux just by switching out the Registry with something else.